web3/contract.py

```diff
@@ -764,9 +765,48 @@ def __prepared_function(self, **kwargs):
                     "The only allowed keyword arguments are: %s" % self.ALLOWED_MODIFIERS)
         else:
             raise TypeError("Use up to one keyword argument, one of: %s" % self.ALLOWED_MODIFIERS)
-        contract_modifier_func = getattr(self.__contract, modifier)
+        contract_modifier_func = getattr(self._contract, modifier)
         return getattr(contract_modifier_func(modifier_dict), self.__function)
 
+class ImplicitContract(ConciseContract):
+    '''
+    ImplicitContract class is similar to the ConciseContract class
+    however it performs a transaction instead of a call if no modifier
+    is given and the method is not marked 'constant' in the ABI.
+
+    The transaction will use the default account and does not send ETH.
+
+    This call
+
+    > contract.withdraw(amount)
+
+    is equivalent to this call in the classic contract:
+
+    > contract.transact({'from': eth.defaultAccount, 'value': 0, ...}).withdraw(amount)
+    '''
+    def __getattr__(self, attr):
+        # If method_name.constant is true in the abi, then call by default
+        call_by_default = \
```

Comment:
This is too difficult to parse/read.  I would recommend breaking this out into a small utility function which **should** also make it easier to test the abi inspection logic directly.

Alternatively you could break this up some across multiple lines, but I think this would benefit from being a stand-alone easy-to-test function.

Reply:
`_find_matching_fn_abi` makes this cleaner. I separated it out